### ISSUE TITLE
Renaming a book with blank fields creates a "unhandled exception"

### ISSUE THREAD

**Issue by DarkMatterMatt** [benchen71/epub-metadata-editor#30]
```
Index was outside the bounds of the array.

************** Exception Text **************
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at EPubMetadataEditor.Form1.LinkLabel3_LinkClicked(Object sender, LinkLabelLinkClickedEventArgs e)
   at System.Windows.Forms.LinkLabel.OnMouseUp(MouseEventArgs e)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.Label.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```
Obviously everything works correctly when the fields are filled as they're supposed to be.
I usually process files with a series index and rename them using it, but occasionally I process a standalone book and don't bother to change the naming template. Instead of giving an error the program should either tell the user *before* renaming, or ignore the error (it currently renames correctly then gives an error).

Great work with making this, it always "just works" and is simple and handy for my OCD regarding metadata, thanks 🎉

**Comment by benchen71** [comment]
I can't reproduce the error.  Can you provide the EPUB which caused the error?  (Rename the file extension to .jpg and you should be able to attach it to this bug report.)  Also, can you please supply the renaming template.  Thanks!

**Comment by DarkMatterMatt** [comment]
https://youtu.be/93hmrEvvh28

**Comment by DarkMatterMatt** [comment]
![bookshots - cross kill epub](https://user-images.githubusercontent.com/15150339/33117643-14eeef8a-cfce-11e7-8737-561c1b061485.jpg)

**Comment by benchen71** [comment]
OK, I could reproduce the error.  I'll try and do the debugging tomorrow.

**Comment by benchen71** [comment]
Fixed in 1.6.4.  Thanks for reporting the bug!

**Comment by DarkMatterMatt** [comment]
Cool, thank *you* for fixing the bug so quickly 😄
